runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -3928,11 +3930,11 @@ void send_polling_response(void* src_addr, c_nodeid_t locale, void* tgt_addr,
     static gni_post_descriptor_t polling_post_descs[PPDESCS_CNT] = { 0 };
     static int last_ppdi = 0;
     int ppdi;
-
+    int spins = 0;
```

Comment:
`send_polling_response()` is always called from a thread.  That thread is the Active Message handler, which handles inbound AMs from remote nodes.  The thread may have executed Chapel code but it has never and will never host a task.  (If it does execute Chapel code it's for a so-called "fast executeOn", where we know the body of a Chapel on-stmt does not do anything that might block or even take very long to execute.)  Also note that `local_yield()` when called by the AM handler will always call `sched_yield()` rather than `chpl_task_yield()`.  Given these circumstances, will there ever be anything for the QSBR checkpoint to do in this case?  (I ask because the latency of this yield plays a role in determining the latency of AM handling and thus the cost of executeOn.)

Reply:
So in this case for "fast executeOn" it depends on a few things. Even if we know privatization stuff is never called inside of one of these statements, it is possible for it to access user protected data, making it potentially unsafe. I mean if its spinning most of the time handling communication we should allow it to keep its checkpoints and just have it register itself.